azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -60,14 +61,21 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
         directly.
 
         """
-        try:
-            uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
+        uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
+
+        for _ in range(1, 6):
```

Comment:
Anyone have any strong feelings about where to put a constant with the number of retries?

Reply:
I'd add as a private member of the GoalState class, or if we want all constants at the top of the file as a private of the module

Reply:
I'm leaning towards constant of the file

Reply:
sounds good